### PR TITLE
fuzz: explicitly cap the vsize of RBFs for diagram checks

### PR DESCRIPTION
**Pull request by instagibbs**
In master we are hitting a case where vsize transactions much larger than max standard size are causing an overflow in not-yet-exposed RBF diagram checking code: https://github.com/bitcoin/bitcoin/pull/29757#issuecomment-2049220195

`ConsumeTxMemPoolEntry` is creating entries with tens of thousands of sigops cost, causing the resulting RBFs to be "overly large".

To fix this I cause the fuzz test to stop adding transactions to the mempool when we reach a potential overflow of `int32_t`.

### CODE REVIEW

File: src/test/fuzz/rbf.cpp [hunk 1]
```diff
@@ -23,7 +23,15 @@ namespace {
 const BasicTestingSetup* g_setup;
 } // namespace
 
+// Maximum number of things we will effect during an RBF
```
Comment (instagibbs) [inline src/test/fuzz/rbf.cpp:26]: this is hopefully a conservative over-estimate since clusters should only be 101kvB total each

**Comment by instagibbs** [comment]
> there is no strict need to tie it to standard sizes.

Pretty sure we can overflow if we allow ~1MvB transactions, as `10,000 * 1,000,000 > 2^31`

Made a more direct limiting attempt. I will squash if it is preferred

**Review by glozow — COMMENTED** [review]
Approach of breaking when adding another tx would overflow seems fine to me. Did you forget to squash?

File: src/test/fuzz/rbf.cpp [hunk 2]
```diff
@@ -130,21 +128,21 @@ FUZZ_TARGET(package_rbf, .init = initialize_package_rbf)
 
         mempool_txs.emplace_back(*parent);
         const auto parent_entry = ConsumeTxMemPoolEntry(fuzzed_data_provider, mempool_txs.back());
-        // We don't want to test things we won't add to mempool (which may cause overflow in size)
-        if (parent_entry.GetTxSize() >= MAX_STANDARD_TX_SIZE) {
+        running_vsize_total += parent_entry.GetTxSize();
+        if (running_vsize_total > std::numeric_limits<int32_t>::max()) {
             return;
```
[the remark this answers is not in the document]
Comment (instagibbs) [reply src/test/fuzz/rbf.cpp]: done

**Comment by instagibbs** [comment]
@glozow was waiting for positive feedback. Squashed.

**Comment by instagibbs** [comment]
pushed a fixup since I think it could have resulted in UB when selecting direct conflicts from the `mempool_txs` vector.

**Review by glozow — COMMENTED** [review]
ACK 016ed248ba0ae64e3f0c93bb47a2cd9b5e49cd85

**Review by marcofleon — COMMENTED** [review]
I ran into this bug when I first started fuzzing `package_rbf` a couple weeks ago... was unsure of what to do about it at the time. But glad to see it addressed here.

ACK 016ed248ba0ae64e3f0c93bb47a2cd9b5e49cd85. I ran libFuzzer on `package_rbf` on the current master branch until the overflow was encountered. Then I built the PR branch and ran the fuzzer using the crash input.
```bash
FUZZ=package_rbf ./src/test/fuzz/fuzz ../crash-d96abc99c0e424a47aa8f4080040e1bc3e3851f7
```
No crash.
```bash
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 1016851959
INFO: Loaded 1 modules   (1208798 inline 8-bit counters): 1208798 [0x107caa4c0, 0x107dd169e), 
INFO: Loaded 1 PC tables (1208798 PCs): 1208798 [0x107dd16a0,0x109043480), 
./src/test/fuzz/fuzz: Running 1 inputs 1 time(s) each.
Running: ../crash-d96abc99c0e424a47aa8f4080040e1bc3e3851f7
Executed ../crash-d96abc99c0e424a47aa8f4080040e1bc3e3851f7 in 1 ms
***
*** NOTE: fuzzing was not performed, you have only
***       executed the target code on a fixed set of inputs.
***
```
I've also been running libFuzzer on `package_rbf` for a while now and the overflow bug hasn't come up.

**Pull request merged**